crates/fuel-streams-core/src/nats/store.rs

```diff
@@ -0,0 +1,90 @@
+use std::fmt::Debug;
+
+use async_nats::jetstream::{
+    consumer::{FromConsumer, IntoConsumerConfig},
+    kv::Watch,
+};
+use async_trait::async_trait;
+use fuel_streams_macros::subject::IntoSubject;
+
+use super::{NatsClient, NatsNamespace, StoreError};
+use crate::nats::{types::*, NatsError};
+
+#[async_trait]
+pub trait Storable: Debug + Clone + serde::Serialize {
```

Comment:
`Streamability` was a concern of `fuel-streams-core`, not `nats`. That's why I mentioned taking these traits, and subjects to the root level. So, renaming it to `Storable` just because we are using the kv store internally is a leak.